### PR TITLE
Add Dashboard Page [2/3] - Caller and SMS Sender Panel

### PR DESCRIPTION
**Pull request by codemist**
# New feature description

This adds the Callers and SMS Senders log subpage/view that the user can navigate to within the Phone Dashboard UI.

image

# How to test

On the Phone Dashboard page, select 'Caller and SMS senders' on the top right of the dashboard card and a list of senders should show. Check that each sender can be individually blocked by clicking its 'Block' button, messages should not be able to through this number if it's in a blocked state.

# Remaining to-do
- [x] Place strings in `pendingTranslations.ftl`
- [x] Add call/text type UI icon indicator (waiting on https://github.com/mozilla/fx-private-relay/pull/2346)
- [x] Microinteractions: hover/focus states
- [x] Mobile UI
- [ ] Panel view when Caller & SMS Senders is disabled (will open up another PR for this one)

# Checklist

- [x] All acceptance criteria are met.
- [x] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Review by groovecoder — CHANGES_REQUESTED** [review]
Looking really good! Just a couple changes requested.

The spot-check works with real data in my local setup! 

image


It felt a bit weird that I needed to refresh the `/phones` page to see new callers and senders show up, but maybe I'm just too used to React auto-magically updating all the UI that I forget it can't detect when something happened ONLY on the server side. (can it?)

File: frontend/src/hooks/api/inboundContact.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { SWRResponse } from "swr";
+import { apiFetch, useApiV1 } from "./api";
+
+export type InboundContact = {
+  id: number;
+  relay_number: number;
+  inbound_number: string;
+  last_inbound_date: string;
```
[the remark this answers is not in the document]
Comment (codemist) [reply frontend/src/hooks/api/inboundContact.ts:8]: Done

File: frontend/src/components/phones/dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -172,11 +188,116 @@ export const PhoneDashboard = () => {
...
+      );
+    });
+
+  const callerSMSSendersPanel = (
+    <div id="secondary-panel" className={styles["dashboard-card"]}>
+      <div className={styles["dashboard-card-caller-sms-senders-header"]}>
+        <span>
+          <button
+            type="button"
+            onClick={showCallerSMSSendersPanel}
```
[the remark this answers is not in the document]
Comment (codemist) [reply frontend/src/components/phones/dashboard/Dashboard.tsx]: Oof, good call out

File: frontend/src/components/phones/dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -172,11 +188,116 @@ export const PhoneDashboard = () => {
...
+        moment(b.last_inbound_date).unix() - moment(a.last_inbound_date).unix()
+    )
+    .map((data) => {
+      return (
+        <li
+          key={data.id}
+          className={data.blocked ? styles["greyed-contact"] : ""}
+        >
+          <span className={styles["sender-number"]}>
+            {formatPhoneNumberToUSDisplay(data.inbound_number)}
```
[the remark this answers is not in the document]
Comment (codemist) [reply frontend/src/components/phones/dashboard/Dashboard.tsx:233]: I can add that in a different PR, when I added the `Moment` library it did a huge `package-lock.json` update so I'm guessing it'll be similar for `LibPhoneNumber

File: frontend/src/components/phones/dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -172,11 +188,116 @@ export const PhoneDashboard = () => {
...
+          </span>
+          <span className={styles["sender-date"]}>
+            <Moment calendar={calendarStrings}>{data.last_inbound_date}</Moment>
+          </span>
+          <span className={styles["sender-controls"]}>
+            <button
+              onClick={() =>
+                inboundContactData.setForwardingState(!data.blocked, data.id)
+              }
+              //onClick={toggleInboundContactForwarding(data.blocked, data.id)}
```
[the remark this answers is not in the document]
Comment (codemist) [reply frontend/src/components/phones/dashboard/Dashboard.tsx]: Done

File: frontend/src/components/phones/dashboard/Dashboard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (codemist) [reply frontend/src/components/phones/dashboard/Dashboard.tsx]: Fixed

**Review by lloan — APPROVED** [review]
LGTM - only comment I have is about the process of picking up new libraries, such as moment. Do we have a process on how we vet and approve libraries @groovecoder @Vinnl? I only ask because I know we're in the security and privacy field and want to make sure we don't start accumulating a load of dependencies on 3rd party libraries if we can. If we have to, it would be great to know that we did our due diligence in vetting out libraries and that we are cognizant that we have to keep them updated.

File: frontend/src/components/phones/dashboard/Dashboard.tsx [hunk 5]
```diff
@@ -172,11 +188,133 @@ export const PhoneDashboard = () => {
...
+        {inboundContactArray}
+      </ul>
+    </div>
+  );
+
+  return (
+    <main className={styles["main-phone-wrapper"]}>
+      {showingPrimaryDashboard ? (
+        // Primary Panel
+        <>{primaryPanel}</>
```
Comment (lloan) [inline frontend/src/components/phones/dashboard/Dashboard.tsx:313]: question (non-blocking): Why are we using fragments to hold `primaryPanel` - same question with ln 317 `callerSMSSendersPanel`

**Pull request merged**